### DISCUSSION TITLE
Support for Maps and Sets

### DISCUSSION THREAD

**Discussion by ivancuric** [pmndrs/valtio#114] [Q&A] [answered]
Does valtio support Maps and Sets?

Asking due to [this discussion](https://www.reddit.com/r/reactjs/comments/kx5sbv/store_a_beautifully_simple_state_management/gjalvu6?utm_source=share&utm_medium=web2x&context=3):

> It's similar yeah. I would guess that Store deals with proxies better though, the library I wrote for that alone weights like double what Valtio weighs, and that's because it may support more built-ins (Maps, Sets etc.) and it ships which optimized comparators, rather than a general deep comparator, for almost every native methods those built-ins have, which improves performance dramatically when working with large objects where deep comparisons can get really expensive.

**Comment by dai-shi** [comment, marked as answer]
Valtio doesn't track mutations inside Maps and Sets. It would require lots of efforts.
Valtio's focus is in creating proxy snapshots which is important for concurrent mode and hook-based render optimization.

**Reply by ivancuric** [reply]
Got it. Wasn't aware that they behaved any different to objects or arrays when it comes to proxies.

**Comment by AjaxSolutions** [comment]
>>> Does valtio support Maps and Sets?

You can use MobX if you need to support Maps and Sets.  I agree that Valtio should be kept simple.

**Comment by dai-shi** [comment]
We have `Set` support by emulation in https://github.com/pmndrs/valtio/releases/tag/v1.2.9

**Reply by dai-shi** [reply]
And we now have `Map` too!

**Reply by dai-shi** [reply]
Related issues & discussions: #266 #272
